pkg/querier/http.go

```diff
@@ -47,12 +47,16 @@ type QueryResponse struct {
 	Result     parser.Value     `json:"result"`
 }
 
+type Engine interface {
+	Query(logql.Params) logql.Query
+}
```

Comment:
This seemed cleaner. The querier defines the interface it wants to receive. What do you think @dannykopping